roachtest: fail test if crdb_internal.check_consistency finds problems

tbg opened the pull request:
This can give us confidence that we'll catch any instances of replica
divergence occurring in roachtest as long as the divergence affects the
MVCCStats keys. This has been the case in all but one reported such issues.

Running the full check is prohibitive and will likely crash the cluster due
to a lack of admission control.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/36241)
<!-- Reviewable:end -->

tbg commented:
(Definitely going to run this against a few choice roachtests before I merge it to make sure it behaves as expected)

tbg commented:
Rebased on top of #36320, PTAL.

andreimatei reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/36241#-:-LbOesFk4n89WhE3Txd5:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei)



<!-- Sent from Reviewable.io -->

tbg commented:
TFTR! Rebased on your PR.

bors r=andreimatei

tbg committed:
roachtest: fail test if crdb_internal.check_consistency finds problems

This can give us confidence that we'll catch any instances of replica
divergence occurring in roachtest as long as the divergence affects
the MVCCStats keys. This has been the case in all but one reported
such issues.

Running the full check is prohibitive and will likely crash the
cluster due to a lack of admission control at this point, but
if we improve its underlying generator to run only N checks in
parallel it should work just fine, though it'll take a long time.

Release note: None
